sdk/servicebus/azure-servicebus/azure/servicebus/management/_utils.py

```diff
@@ -44,3 +60,110 @@ def get_next_template(list_func, *args, **kwargs):
             )
         )
     return feed_element
+
+
+def deserialize_value(value, value_type):
+    if value_type in ("int", "long"):
+        value = int(value)
+    elif value_type == "boolean":
+        value = value.lower() == "true"
+    elif value_type == "double":
+        value = float(value)
+    elif value_type == "datetime":
+        value = isodate.parse_datetime(value)
+    elif value == "duration":
+        value = isodate.parse_duration(value)
+    return value
+
+
+def serialize_value_type(value):
+    value_type = type(value)
+    if value_type == str:
+        return "string", value
+    if value_type == int:
+        return "int" if value <= constants.INT_MAX_VALUE_CSHARP else "long", str(value)
+    if value_type == float:
+        return "double", str(value)
+    if value_type == bool:
+        return "boolean", str(value)
+    if value_type == datetime:
+        return "datetime", isodate.datetime_isoformat(value)
+    if value_type == timedelta:
+        return "duration", isodate.duration_isoformat(value)
+    raise ValueError("value {} of type {} is not supported for the key value".format(value, value_type))
+
+
+def deserialize_key_values(root, parameters):
```

Comment:
Out variables are less typical in my experience.  Would do my typical "kindly put a small docstring here noting the behavior/parameters" rather than have you rejigger things too much. (Can even justify this via the current push to document internal code as well :P)

Reply:
I changed param names to (xml_parent, key_values) and added docstring to explain `deserialize_key_values` and `serialize_key_values`

Reply:
LGTM.  deserialize_rule_key_values was elided but is that one of the ones that's "going away post-autorest-fixes"?  if not I might do for completeness.

Reply:
It's going away post-autorest-fixes.  It has docstring about the autorest issue too.